lighthouse-core/gather/driver.js

```diff
@@ -793,6 +835,19 @@ class Driver {
     // CPU listener. Resolves when the CPU has been idle for cpuQuietThresholdMs after network idle.
     let waitForCPUIdle = this._waitForNothing();
 
+    const waitForSecurityCheck = this._waitForSecurityCheck();
+    const securityCheckPromise = waitForSecurityCheck.promise.then(explanations => {
+      return function() {
+        const insecureDescriptions = explanations
+          .filter(exp => exp.securityState === 'insecure')
+          .map(exp => exp.description);
+        const err = new LHError(LHError.errors.INSECURE_DOCUMENT_REQUEST, {
+          securityMessages: insecureDescriptions.join(' '),
+        });
+        throw err;
+      };
+    });
```

Comment:
ha, I think I disagree with @patrickhulce. The internals of `SecurityStateExplanation` seem like a distraction from the actual purpose of `_waitForFullyLoaded`, since either way we're using `waitForSecurityCheck` to alert us of an error, the question is if we should do more work in there or out here.

If that's not convincing, maybe we could split the difference? Filter and map within `_waitForSecurityCheck` and then it only needs to be 
```js
const securityCheckPromise = waitForSecurityCheck.promise.then(securityMessages => {
  return function() {
    throw new new LHError(LHError.errors.INSECURE_DOCUMENT_REQUEST, {securityMessages});
  }
});
```

(using the cleanupFn to reject the promise is a little weird but w/e :P)